contracts/helpers/TransferHelper.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.7;
+
+import "./TransferHelperStructs.sol";
+
+import { TokenTransferrer } from "../lib/TokenTransferrer.sol";
+
+import { ConduitInterface } from "../interfaces/ConduitInterface.sol";
+
+// prettier-ignore
+import {
+    ConduitControllerInterface
+} from "../interfaces/ConduitControllerInterface.sol";
+
+import { Conduit } from "../conduit/Conduit.sol";
+
+import { ConduitTransfer } from "../conduit/lib/ConduitStructs.sol";
+
+// prettier-ignore
+import {
+    TransferHelperInterface
+} from "../interfaces/TransferHelperInterface.sol";
+
+/**
+ * @title TransferHelper
+ * @author stuckinaboot, stephankmin
+ * @notice TransferHelper is a utility contract for transferring
+ *         ERC20/ERC721/ERC1155 items in bulk to a specific recipient.
+ */
+contract TransferHelper is TransferHelperInterface, TokenTransferrer {
+    // Allow for interaction with the conduit controller.
+    ConduitControllerInterface internal immutable _CONDUIT_CONTROLLER;
+
+    // Cache the conduit creation hash used by the conduit controller.
+    bytes32 internal immutable _CONDUIT_CREATION_CODE_HASH;
+
+    /**
+     * @dev Set the supplied conduit controller and retrieve its
+     *      conduit creation code hash.
+     *
+     *
+     * @param conduitController A contract that deploys conduits, or proxies
+     *                          that may optionally be used to transfer approved
+     *                          ERC20/721/1155 tokens.
+     */
+    constructor(address conduitController) {
+        // Get the conduit creation code hash from the supplied conduit
+        // controller and set it as an immutable.
+        ConduitControllerInterface controller = ConduitControllerInterface(
+            conduitController
+        );
+        (_CONDUIT_CREATION_CODE_HASH, ) = controller.getConduitCodeHashes();
+
+        // Set the supplied conduit controller as an immutable.
+        _CONDUIT_CONTROLLER = controller;
+    }
+
+    /**
+     * @notice Transfer multiple items to a single recipient.
+     *
+     * @param items      The items to transfer.
+     * @param recipient  The address the items should be transferred to.
+     * @param conduitKey The key of the conduit through which the bulk transfer
+     *                   should occur.
+     *
+     * @return magicValue A value indicating that the transfers were successful.
+     */
+    function bulkTransfer(
+        TransferHelperItem[] calldata items,
+        address recipient,
```

Comment:
Maybe verify that the `recipient` is able to receive NFT's (in case its a contract)